public/app/plugins/datasource/mysql/specs/datasource.test.ts

```diff
@@ -25,12 +24,13 @@ describe('MySQLDatasource', () => {
       jsonData: {
         defaultProject: 'testproject',
       },
-    } as unknown as DataSourceInstanceSettings<MySQLOptions>;
+    } as unknown as DataSourceInstanceSettings<SQLOptions>;
     const templateSrv: TemplateSrv = new TemplateSrv();
     const variable = { ...initialCustomVariableModelState };
     fetchMock.mockImplementation((options) => of(createFetchResponse(response)));
 
-    const ds = new MysqlDatasource(instanceSettings, templateSrv);
+    const ds = new MySqlDatasource(instanceSettings);
```

Comment:
I think this was good. You just need to pass the templateSrv to the super call.

Reply:
why is it necessary?

Reply:
This is only to make the test work without the reflect.